src/analysisd/eventinfo.c

```diff
@@ -75,22 +76,74 @@ Eventinfo *Search_LastSids(Eventinfo *my_lf, RuleInfo *rule, __attribute__((unus
         /* Check for same ID */
         if (rule->context_opts & SAME_ID) {
             if ((!lf->id) || (!my_lf->id)) {
-                continue;
+                goto next_it;
             }
 
             if (strcmp(lf->id, my_lf->id) != 0) {
-                continue;
+                goto next_it;
             }
         }
 
         /* Check for repetitions from same src_ip */
         if (rule->context_opts & SAME_SRCIP) {
             if ((!lf->srcip) || (!my_lf->srcip)) {
-                continue;
+                goto next_it;
             }
 
             if (strcmp(lf->srcip, my_lf->srcip) != 0) {
-                continue;
+                goto next_it;
+            }
+        }
+
+        /* Check for repetitions from same dynamic fields */
+        if (rule->context_opts & SAME_FIELD) {
+            if (my_lf->nfields == 0 || lf->nfields == 0)
+                goto next_it;
+
+            i = 0;
+            while (rule->same_fields[i]) {
+                found = 0;
+                for (j = 0; j < my_lf->nfields; j++) {
```

Comment:
Please optimize this algorithm: each `for` will iterate over all the fields even after matching a field.
Surely you can use `break` and `found` to achieve it.